### PR TITLE
Implement company Ai end function

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Split off from #2189

Doing this standalone pointed out a bug in the ai in OpenLoco where bankrupt ai would never get removed!

### CODE REVIEW

File: src/OpenLoco/src/World/CompanyAi.cpp [hunk 1]
```diff
@@ -224,7 +229,7 @@ namespace OpenLoco
             bool hasAssets = false;
             for (auto& thought : company.aiThoughts)
             {
-                if (thought.type == AiThoughtType::null)
+                if (thought.type != AiThoughtType::null)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/World/CompanyAi.cpp:232]: woops

File: src/OpenLoco/src/World/CompanyAi.cpp [hunk 2]
```diff
@@ -812,12 +816,102 @@ namespace OpenLoco
...
+                auto* elRoad = el.as<World::RoadElement>();
+                if (elTrack != nullptr)
+                {
+                    if (elTrack->owner() != id)
+                    {
+                        continue;
+                    }
+                    reTryTile = true;
+
+                    GameCommands::TrackRemovalArgs args{};
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/CompanyAi.cpp:840]: Considering the two game commands are so alike, perhaps you extract getting track/road arguments into a templated function? That should reduce/remove the duplication in the two blocks here, and make the function more readable.

**Comment by AaronVanGeffen** [comment]
> Doing this standalone pointed out a bug in the ai in OpenLoco where bankrupt ai would never get removed!

Nice catch! Could you add a changelog entry for this?

File: src/OpenLoco/src/World/CompanyAi.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/World/CompanyAi.cpp:840]: On second thought, the field names are slightly different, making this more complicated than I originally thought. I don't want to move the goal posts too much, so let's just leave this be.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**